include/kernel.h

```diff
@@ -32,6 +36,7 @@ void set_timer();
 void disable_interrupts();
 void enable_interrupts();
 void set_mie(unsigned int value);
+void set_mtvec(void *ptr);
```

Comment:
Reading through other implementations in C I had hard time with processor specific terms like MTVEC and MIE. It requires one to read "assembly" level documentation which IMO partially defeats the purpose of C.

I was thinking, maybe write it in full:
set_mtvec() --> set_machine_trap_vector(void* mtvec); // note I would use "mtvec" as the name of the argument, so it is easier to match it with the docs.
set_mie() --> set_machine_interrupts(unisgned int mie)

Or maybe have 2 levels for readability, one low-level that uses machine level terms that map directly to RISCV documentation and one that has human readable names, but calls into low-level:
void set_mie(unsigned int value);
...
void set_machine_interrupts(unsigned int mie) { set_mie(value); }

Reply:
That's a very good point. I was having in mind to do something like the second approach soon: a lower level, machine-specific layer, wrapped by something more readable. I will move all the names containing register names to a separate `riscv.c` file as the first obvious step, and your suggestion to spell out full names seems like a good idea to try for the wrapping layer.